apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,408 @@
+        /// </summary>
+        /// <returns>A canonical string representation of this date.</returns>
+        public override string ToString() => ToString((string)null);
+
+        /// <summary>
+        /// Returns a canonical string representation of the Date represented by this instance in the specified format.
+        /// </summary>
+        /// <param name="format">The format for Date value.</param>
+        /// <returns>
+        /// A <see cref="System.String" /> that represents this instance.
+        /// </returns>
+        public string ToString(string format) => ToString(format, null);
+
+        /// <summary>
+        /// Returns a canonical string representation of the Date represented by this instance in the specified format.
+        /// </summary>
+        /// <param name="formatProvider">The format provider.</param>
+        /// <returns>
+        /// A <see cref="System.String" /> that represents this instance.
+        /// </returns>
+        public string ToString(IFormatProvider formatProvider) => ToString(null, formatProvider);
+
+        /// <summary>
+        /// Returns a canonical string representation of the Date represented by this instance in the specified format.
+        /// </summary>
+        /// <param name="format">The format.</param>
+        /// <param name="formatProvider">The format provider.</param>
+        /// <returns>
+        /// A <see cref="System.String" /> that represents this instance.
+        /// </returns>
+        public string ToString(string format, IFormatProvider formatProvider)
+        {
+            // TODO: Implement proper format and formatProvider implementation based on discussion with Jon and Amanda.
+            // What if someone provides g or G or s or any other formats like hh:mm:ss which displays time as well?
+            // Do we want to expose flexibility of format here or keep it simple as we won't be able to validate the format?
+            return string.IsNullOrWhiteSpace(format) || (format.Length >= 1 && (format[0] == 'g' || format[0] == 'G')) ? ToDateTime().ToString("yyyy-MM-dd", formatProvider) : ToDateTime().ToString(format, formatProvider);
+        }
+
+        /// <summary>
+        /// Returns a SpannerDate instance that is set to the date part of the specified DateTime.
+        /// </summary>
+        /// <param name="dateTime">The date time.</param>
+        /// <returns>The SpannerDate instance</returns>
+        public static SpannerDate FromDateTime(DateTime dateTime) => new SpannerDate(dateTime);
+
+        /// <summary>
+        /// Creates a new instance of the SpannerDate structure to the specified number of days.
+        /// </summary>
+        /// <param name="dayNumber">The day number.</param>
+        /// <returns></returns>
+        public static SpannerDate FromDayNumber(int dayNumber) => new SpannerDate(dayNumber);
+
+        /// <summary>
+        /// Converts value of SpannerDate to The DateTime.
```

Comment:
Nit: capital T here and below.

I think we also need to document the Kind returned (which I assume is Unspecified)